src/materials/ConcreteExpansionEigenstrainBase.C

```diff
@@ -0,0 +1,257 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/*                       BlackBear                              */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "ConcreteExpansionEigenstrainBase.h"
+#include "RankTwoTensor.h"
+
+template <>
+InputParameters
+validParams<ConcreteExpansionEigenstrainBase>()
+{
+  InputParameters params = validParams<ComputeEigenstrainBase>();
+  MooseEnum expansion_type("Isotropic Anisotropic");
+  params.addParam<MooseEnum>(
+      "expansion_type", expansion_type, "Type of expansion resulting from volumetric strain");
+  params.addRequiredRangeCheckedParam<Real>(
+      "compressive_strength", "compressive_strength > 0", "Compressive strength of concrete");
+  params.addRequiredRangeCheckedParam<Real>(
+      "expansion_stress_limit",
+      "expansion_stress_limit > 0",
+      "Upper bound compressive stress beyond which no expansion occurs");
+  params.addRequiredRangeCheckedParam<Real>(
+      "tensile_strength", "tensile_strength > 0", "Tensile strength of concrete");
+  return params;
+}
+
+ConcreteExpansionEigenstrainBase::ConcreteExpansionEigenstrainBase(
+    const InputParameters & parameters, const std::string volumetric_expansion_name)
+  : ComputeEigenstrainBase(parameters),
+    _expansion_type(getParam<MooseEnum>("expansion_type").getEnum<ExpansionType>()),
+    _f_compress(getParam<Real>("compressive_strength")),
+    _f_u(getParam<Real>("expansion_stress_limit")),
```

Comment:
I think this variable would be a strong candidate for renaming. Maybe something like `expansion_stress_limit`?

Reply:
I renamed to _sigma_u to match the notation in the paper.